src/client/context.cc

```diff
@@ -359,6 +358,26 @@ void ClientContext::UpdateClientTunnel(
       rebind =
           (client_tunnel->GetEndpoint()
            != boost::asio::ip::tcp::endpoint(next_addr, tunnel.port));
+    if (!rebind)  // check for change in destination
```

Comment:
Can we add a quick note here about *why* we want to rebind when these tunnel attributes change (e.g., a quick/easy way to update attributes even if local address/port hasn't changed)?